src/monster.cpp

```diff
@@ -684,13 +685,28 @@ int monster::hp_percentage() const
 
 void monster::process_triggers()
 {
- anger += trigger_sum(&(type->anger));
- anger -= trigger_sum(&(type->placate));
- if (morale < 0) {
-  if (morale < type->morale && one_in(20))
-  morale++;
- } else
-  morale -= trigger_sum(&(type->fear));
+    anger += trigger_sum( &(type->anger) );
+    anger -= trigger_sum( &(type->placate) );
+    morale -= trigger_sum( &(type->fear) );
+    if( morale != type->morale && one_in( 10 ) ) {
+        if( morale < type->morale ) {
+            morale++;
+        } else {
+            morale--;
+        }
+    }
+
+    if( anger != type->agro && one_in( 10 ) ) {
+        if( anger < type->agro ) {
+            anger++;
+        } else {
+            anger--;
+        }
+    }
+
+    // Cap values at [-100, 100] to prevent perma-angry moose etc.
+    morale = std::min( 100, std::max( 0, morale ) );
```

Comment:
From the comment (and from the PR description), this should be `std::max( -100` to allow negative values.